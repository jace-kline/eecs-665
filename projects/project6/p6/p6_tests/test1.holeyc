int g;
int a(int b, int c) {
    if (b == c) {
        return 0;
    } else {
        b = b - 1 - c;
    }
    return b;
}