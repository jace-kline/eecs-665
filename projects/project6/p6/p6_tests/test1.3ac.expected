[BEGIN GLOBALS]
g
[END GLOBALS]
[BEGIN a LOCALS]
b (formal)
c (formal)
tmp0 (tmp)
tmp1 (tmp)
tmp2 (tmp)
[END a LOCALS]
lbl_fun_a:  enter a
            getarg 1 [b]
            getarg 2 [c]
            [tmp0] := [b] EQ64 [c]
            IFZ [tmp0] GOTO lbl_1
            setret 0
            goto lbl_0
            goto lbl_2
lbl_1:      nop
            [tmp1] := [b] SUB64 1
            [tmp2] := [tmp1] SUB64 [c]
            [b] := [tmp2]
lbl_2:      nop
            setret [b]
            goto lbl_0
lbl_0:      leave a